### PR TITLE
refactor(server): job repository

### PR DESCRIPTION
**Pull request by jrasm91**
This PR replaces bull queues in all of the services with `IJobRepository`, preparing to move them into domain. (net **-300** lines of code :smile:)

Merge after #1353.

### CODE REVIEW

File: docker/.env.test [hunk 1]
```diff
@@ -5,7 +5,7 @@ DB_PASSWORD=postgres
 DB_DATABASE_NAME=e2e_test
 
 # Redis
-REDIS_HOSTNAME=immich_redis_test
+REDIS_HOSTNAME=immich-redis-test
```
Comment (samip5) [inline docker/.env.test:8]: Are you sure this was intended change?

Comment (jrasm91) [reply docker/.env.test:8]: That's the hostname in the container for the test env and the e2e tests was running for 5 hours because it was wrong 😁

Comment (samip5) [reply docker/.env.test:8]: Ah, okay. :D

**Comment by alextran1502** [comment]
#1353 is merged. Can you resolve conflict?

**Pull request closed without merging**